### PR TITLE
Migrate Schneider dimmer quirks to v2, add missing models

### PR DESCRIPTION
**Pull request by uvNikita**
## Proposed change
I migrated quirks to v2 and added models other models from the spec.
Looking through the spec docs, they all seem to have equivalent functionality.
The original quirk was missing SEOnOff cluster, so I added it too.

## Additional information
Specs:
[ZB Spec - Push-button Dimmer - 110422.pdf](https://github.com/user-attachments/files/17831848/ZB.Spec.-.Push-button.Dimmer.-.110422.pdf)
[ZB Spec - Rotary Dimmer - 110422.pdf](https://github.com/user-attachments/files/17831849/ZB.Spec.-.Rotary.Dimmer.-.110422.pdf)
[ZB Spec - Relay Switch 10A - 110422.pdf](https://github.com/user-attachments/files/17832116/ZB.Spec.-.Relay.Switch.10A.-.110422.pdf)

## Checklist

- [x] The changes are tested and work correctly
- [x] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

**Comment by uvNikita** [comment]
Adding the original PRs for the reference #1705 #3111

File: zhaquirks/schneiderelectric/dimmers.py [hunk 1]
```diff
@@ -1,129 +1,33 @@
...
+(
+    QuirkBuilder(SE_MANUF_NAME, "NHROTARY/DIMMER/1")
+    .applies_to(SE_MANUF_NAME, "NHROTARY/UNIDIM/1")
+    .applies_to(SE_MANUF_NAME, "NHPB/DIMMER/1")
+    .applies_to(SE_MANUF_NAME, "NHPB/UNIDIM/1")
+    .replaces(SEBasic, endpoint_id=3)
+    .replaces(SEBallast, endpoint_id=3)
+    .replaces(SEOnOff, endpoint_id=3)
+    .replaces(SEBasic, endpoint_id=21)
+    .adds(SESpecific, endpoint_id=21)
```
Comment (TheJulianJES) [inline zhaquirks/schneiderelectric/dimmers.py]: Shouldn't this also be `.replaces`, as the `SESpecific.cluster_id` was in the signature before?

File: zhaquirks/schneiderelectric/dimmers.py [hunk 2]
```diff
@@ -1,129 +1,33 @@
...
-                ],
-            },
-        }
-    }
+(
+    QuirkBuilder(SE_MANUF_NAME, "NHPB/SWITCH/1")
+    .replaces(SEBasic)
+    .replaces(SEOnOff)
+    .replaces(SEBasic, endpoint_id=21)
+    .adds(SESpecific, endpoint_id=21)
```
Comment (TheJulianJES) [inline zhaquirks/schneiderelectric/dimmers.py]: And same here? Also `.replaces`?

File: zhaquirks/schneiderelectric/dimmers.py (hunk 1 above)
Comment (TheJulianJES) [reply zhaquirks/schneiderelectric/dimmers.py]: Tbh, I'm not entirely sure why we even have `.adds(cluster)`. I think you always want to remove a cluster with the same `cluster_id` when adding/replacing. If it already exists, `.adds(cluster)` won't do anything IIRC.

Comment (TheJulianJES) [reply zhaquirks/schneiderelectric/dimmers.py]:
```suggestion
    .replaces(SESpecific, endpoint_id=21)
```

File: zhaquirks/schneiderelectric/dimmers.py (hunk 2 above)
Comment (TheJulianJES) [inline zhaquirks/schneiderelectric/dimmers.py]:
```suggestion
    .replaces(SESpecific, endpoint_id=21)
```

**Review by TheJulianJES — APPROVED** [review]
Thanks! I've replaced `adds(SESpecific)` with `replaces(SESpecific)`, as we want to ensure the original cluster is removed first (and replaced with our custom implementation).
There's some more info in the review comments above.

**Comment by uvNikita** [comment]
@TheJulianJES I was just assuming `adds` should be used on Manufacturer Specific clusters and `replaces` on the standard ones.

Will keep that in mind for the future, thanks a lot for the review!

**Comment by TheJulianJES** [comment]
Yeah, we need to get some docs together at some point.. I'm not sure if there's ever any reason to use `adds` instead of `replaces`. We should also take a look at that and remove `adds` if needed 😄

**Pull request merged**